karate-core/src/main/java/com/intuit/karate/Main.java

```diff
@@ -110,16 +111,19 @@
     @Option(names = {"-d", "--debug"}, arity = "0..1", defaultValue = "-1", fallbackValue = "0",
             description = "debug mode (optional port else dynamically chosen)")
     int debugPort;
-    
+
     @Option(names = {"-D", "--dryrun"}, description = "dry run, generate html reports only")
-    boolean dryRun;    
+    boolean dryRun;
 
     @Option(names = {"-j", "--jobserver"}, description = "job server url")
     String jobServerUrl;
 
     @Option(names = {"-i", "--import"}, description = "import and convert a file")
     String importFile;
 
+    @Option(names = {"-H", "--hook"}, description = "Class name of a RuntimeHook or RuntimeHookFactory to add")
+    Class<?> hookFactoryClass;
```

Comment:
maybe later, but it may be a good idea to model this as a collection so you can pass multiple hooks